### ISSUE TITLE
Docker macvlan & Virtualbox - Host, Network, Guest, and Container intercommunication

### ISSUE THREAD

**Issue by h6w** [docker/libnetwork#1706]
According to the [current macvlan documentation](https://docs.docker.com/engine/userguide/networking/get-started-macvlan/#macvlan-bridge-mode-example-usage):

> Note: In Macvlan you are not able to ping or communicate with the default namespace IP address. For example, if you create a container and try to ping the Docker host’s eth0 it will not work. That traffic is explicitly filtered by the kernel modules themselves to offer additional provider isolation and security.

I'm questioning this premise.

**Docker**

If I attach a docker network with the macvlan bridge to the host's interface, then the docker container can ping other docker containers, and other devices in the connected network, **_but not the host itself_**.

**VirtualBox**

However, if I have a VirtualBox installation that attaches to the host's interface, I **_am able to ping the host_** from the VirtualBox guest and vice versa and other VirtualBox guests, and other devices in the connected network.

**Together**

Furthermore, if I run a docker network and a VirtualBox network on the same interface, the results are the same.  In addition, the VirtualBox guests and docker containers cannot intercommunicate.

I have tried moving the host IP onto it's own bridge, but that only meant that the VirtualBox guests could no longer communicate with the host.

_**How is it possible to get both VirtualBox guests and Docker containers to communicate with each other and the host and devices in the same network using the same interface?**_

**Comment by h6w** [comment]
VirtualBox uses a netfilter driver to intercept packets at the interface.  This may mean that the VirtualBox netfilter driver is not detecting packets coming from the docker bridge.

**Comment by h6w** [comment]
I have successfully managed to get this working by using OpenVSwitch.

Basically, I created 3 OVS bridges:
phys_br - for the physical interface
vbox_br - for the Virtualbox Machines
dock_br - for the Docker Containers

Then I:
- patched phys_br to vbox_br; and 
- patched phys_br to dock_br, and 
- put the ethernet interface onto the phys_br bridge; and
- made all Virtualbox images use vbox_br; and 
- created a docker network with "-o parent=dock_br"

Also:
I had to run `ethtool -K $IFACE tso off gro off gso off tx off rx off` where $IFACE is phys_br, vbox_br, and dock_br, otherwise I had significant TCP dropouts (whereas ICMP and UDP would work up to this point)

Anyhoo, I have had this working in testing for over a week and now production for more than 24 hours.  So far so good! :-D
